app/src/ui/toolbar/push-pull-button.tsx

```diff
@@ -523,12 +556,31 @@ export class PushPullButton extends React.Component<
         dropdownContentRenderer={this.getDropdownContentRenderer(
           dropdownItemTypes
         )}
+        returnFocusOnDeactivate={this.returnFocusOnDeactivate()}
+        onDropdownFocusTrapDeactivate={this.onDropdownFocusTrapDeactivate}
       >
         {renderAheadBehind(aheadBehind, numTagsToPush)}
       </ToolbarDropdown>
     )
   }
 
+  private returnFocusOnDeactivate = () => {
+    const isForcePushOptionAvailable =
+      this.props.forcePushBranchState !== ForcePushBranchState.NotAvailable
+
+    return (
+      !isForcePushOptionAvailable || !this.props.askForConfirmationOnForcePush
+    )
```

Comment:
The first time I read this (before reading your comments) it was a bit misleading because it's returning `false` when we want to return focus to the previous active element 😂  Could you add a comment here to explain why we want to bypass that behavior from FocusTrap?